core/block.ts

```diff
@@ -95,6 +95,9 @@ export class Block implements IASTNodeLocation, IDeletable {
   /** An optional method called during initialization. */
   init?: (() => AnyDuringMigration)|null = undefined;
 
+  /** An optional method called during disposal. */
+  destroy?: (() => AnyDuringMigration)|null = undefined;
```

Comment:
I went with void & removed the `|null`. I thought `undefined` made more sense as a default value, since this property /is/ optional, and its either defined or its not.